code/modules/food_and_drinks/kitchen_machinery/food_cart.dm

```diff
@@ -10,151 +6,259 @@
 	density = TRUE
 	anchored = FALSE
 	use_power = NO_POWER_USE
-	var/food_stored = 0
-	var/glasses = 0
-	var/portion = 10
-	var/selected_drink
-	var/list/stored_food = list()
+	//Max amount of items that can be in the cart's contents list
+	var/contents_capacity = 80
+	//How many drinking glasses the cart has
+	var/glass_quantity = 10
+	//Max amount of drink glasses the cart can have
+	var/glass_capacity = 30
+	//Max amount of reagents that can be in cart's storage
+	var/reagent_capacity = 200
+	//Sound made when an item is dispensed
+	var/dispense_sound = 'sound/machines/click.ogg'
+	//Sound made when an item is inserted
+	var/insert_sound = 'sound/effects/rustle2.ogg'
+	//Sound made when selecting/deselecting an item
+	var/select_sound = 'sound/machines/doorclick.ogg'
+	//List used to show food items in UI
+	var/list/food_ui_list = list()
+	//List of transfer amounts for reagents
+	var/list/transfer_list = list(5, 10, 15, 20, 30, 50)
+	//What transfer amount is currently selected
+	var/selected_transfer = 0
+	//Mixer for dispencing drinks
 	var/obj/item/reagent_containers/mixer
 
+/obj/machinery/food_cart/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "FoodCart", name)
+		ui.open()
+
+/obj/machinery/food_cart/ui_data(mob/user)
+	//Define data for sending info to UI
+	var/list/data = list()
+
+	//Define lists for each property of data so that they send to UI regardless of what happens
+	//Thanks bug eating lizard and offbeatwitch for helping solve UI updating issue
+	data["food"] = list()
+	data["mainDrinks"] = list()
+	data["mixerDrinks"] = list()
+	data["storage"] = list()
+
+	
+	//Loop through food list for data to send to food tab
+	for(var/item_detail in food_ui_list)
+		//Create needed list and variable for geting data for UI
+		var/list/details = list()
+		var/obj/item/reagent_containers/food/item = new item_detail
+
+		//Get information for UI
+		details["name"] = item.name
+		details["quantity"] = find_amount(item)
+		details["type_path"] = item.type
+
+		//Get an image for the UI
+		var/icon/item_pic = getFlatIcon(item)
+		var/md5 = md5(fcopy_rsc(item_pic))
+		if(!SSassets.cache["photo_[md5]_[item.name]_icon.png"])
+			SSassets.transport.register_asset("photo_[md5]_[item.name]_icon.png", item_pic)
+		SSassets.transport.send_assets(user, list("photo_[md5]_[item.name]_icon.png" = item_pic))
+		details["image"] = SSassets.transport.get_asset_url("photo_[md5]_[item.name]_icon.png")
+
+		//Add to food list
+		data["food"] += list(details)
```

Comment:
I think you can avoid the extra list allocation here using list.Add
```suggestion
		data["food"].Add(details)
```

Reply:
I tried that and it somehow prevents the UI from showing food.